lib/silencer/rails/logger.rb

```diff
@@ -35,17 +35,45 @@ def initialize(app, *args)
       end
 
       def call(env)
-        old_logger_level     = ::Rails.logger.level
-        ::Rails.logger.level = ::Logger::ERROR if silence_request?(env)
+        if silence_request?(env)
+          quiet do
+            super
+          end
+        else
+          super
+        end
+      end
 
-        super
+      private
+      def quiet(&block)
+        if ::Rails.logger.respond_to?(:silence)
+          quiet_with_silence do
+            block.call
```

Comment:
do you need to invoke the block here since it is also invoked within the `quiet_with_silence` method?

Perhaps, i'm looking at this incorrectly, but i'd imagine you could simplify this to `quiet_with_silence(&block)`

quiet_with_silence(block)